### ISSUE TITLE
how could I use dali in the inference of single image?

### ISSUE THREAD

**Issue by CoinCheung** [NVIDIA/DALI#1166]
Hi, 

Thanks for providing this marvelous tool. After seeing the tutorial, I learned that I could use dali to load training set and validation set, but how could I use dali to load single image for inference ? Do I have an option to use dali in the way I use `cv2.imread()` ? Must I use piplines?

**Comment by awolant** [comment]
Hi, thanks for the question. Currently DALI does not support something like this. Underlying execution model relies on the pipeline mechanism.
Workaround is to build and run the pipeline with batch_size=1. Maybe you can combine this with the [ExternalSource operator](https://docs.nvidia.com/deeplearning/sdk/dali-developer-guide/docs/examples/external_input.html) and pass one image at a time to it?

**Comment by CoinCheung** [comment]
Thanks for replying!! 

BTW, do I have other options to read images apart from the `ops.FileReader` method? 
Currently, I have to use `np.from_buffer` to load raw image, and then use nvidia-decoder to decode it. Do I have a method to load an image given its path and load it to the gpu directly?

**Comment by awolant** [comment]
Right now, there is no way of directly load data from the drive to the GPU memory in DALI. All of our readers are using the host memory as a intermediate buffer.
AFAIK, technology to do something like this exists and it's called [GPUDirect](https://developer.nvidia.com/gpudirect), but we are yet to take a closer look at it. If you really want this feature, maybe you can experiment with it yourself. DALI reader with this capability would make an exciting external contribution.

**Comment by JanuszL** [comment]
@CoinCheung - another reason why we cannot load data directly to the GPU is the fact that decoding still uses CPU for the Huffman (as this part cannot be efficiently parallelized with the usual batch size used in DL applications).

**Comment by JanuszL** [comment]
Hi @CoinCheung,

DALI supports GPUDirect inside GPU variant of [the `numpy` reader](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/supported_ops.html#nvidia.dali.fn.readers.numpy).
I also recommend checking [DALI TRITON backend](https://github.com/triton-inference-server/dali_backend) to see how to use DALI in the inference.
